tests/posix/common/src/clock.c

```diff
@@ -109,8 +110,16 @@ void test_posix_realtime(void)
 		/* printk("Delta %d: %lld\n", i, delta); */
 
 		/* Allow for a little drift */
-		zassert_true(error > 90, "Clock inaccurate");
-		zassert_true(error < 110, "Clock inaccurate");
+		/* NOTE: this test is miscalibrated.  It's looking for
+		 * the clock to advance between 90 and 110ms
+		 * (inclusive), but the sleep above is for 90ms.  A
+		 * high tick rate system will nail that delay exactly
+		 * and be right on the edge of a failure.  It should
+		 * be fixed (but not when the timer code is
```

Comment:
> (but not when the timer code is
		 * simultaneously being worked on!).

I feel your pain, in PRs like https://github.com/zephyrproject-rtos/zephyr/pull/16621, https://github.com/zephyrproject-rtos/zephyr/pull/16427, https://github.com/zephyrproject-rtos/zephyr/pull/15937, our communication with guys like @nashif, @mbolivar is centered around me explaining (and them not understanding) that everything can't be fixed at once.

Then, let's still keep that to codereview, not code comments. If the nature of your patch is to replace `<`/`>` with `<=`/`>=` is timing comparison code, please do just that, and open a separate ticket with concerns expressed in code comments here (can be copied verbatim to that ticket).